addons/mail/sfu_server/src/models/channel.js

```diff
@@ -0,0 +1,269 @@
+import { EventEmitter } from "node:events";
+
+import * as config from "#src/config.js";
+import { getAllowedCodecs, Logger } from "#src/utils/utils.js";
+import { AuthenticationError, OvercrowdedError } from "#src/utils/errors.js";
+import { Session, SESSION_CLOSE_CODE } from "#src/models/session.js";
+import { getWorker } from "#src/services/rtc.js";
+
+const logger = new Logger("CHANNEL");
+
+const mediaCodecs = getAllowedCodecs();
+
+/**
+ * @typedef {Object} SessionsStats
+ * @property {{total: number, screen: number, audio: number, camera: number}} incomingBitRate
+ * @property {number} count
+ * @property {number} cameraCount
+ * @property {number} screenCount
+ */
+
+/**
+ * @fires Channel#sessionJoin
+ * @fires Channel#sessionLeave
+ * @fires Channel#close
+ */
+export class Channel extends EventEmitter {
+    /** @type {Map<string, Channel>} */
+    static records = new Map();
+    /** @type {Map<string, Channel>} */
+    static recordsByIssuer = new Map();
+
+    /** @type {string} */
+    createDate;
+    /** @type {string} */
+    remoteAddress;
+    /** @type {string} */
+    uuid;
+    /** @type {string} short of uuid for logging */
+    name;
+    /** @type {import("mediasoup").types.Router}*/
+    router;
+    /** @type {Map<number, Session>} */
+    sessions = new Map();
+    /** @type {import("mediasoup").types.Worker}*/
+    _worker;
+    /** @type {NodeJS.Timeout} */
+    _closeTimeout;
+
+    /**
+     * @returns {Channel}
+     * @param {string} remoteAddress
+     * @param {string} issuer
+     * @param {{ useWebRtc: boolean }} [param2]
+     */
+    static async create(remoteAddress, issuer, { useWebRtc = true } = {}) {
```

Comment:
What is the point of useWebRtc being inside an object?
Can't this just be 
```js
    static async create(remoteAddress, issuer, useWebRtc = true) {
```

Reply:
It's not great to pass a boolean value that cannot be understood by just reading the LOC, without IDE integration.
It's prone to mistakes, in that we always have to guess what this boolean means, and if the definition is changed then we no longer know what it semantically meant. Code could works for a few weeks by chance and break at any time and we waste time because the code is not explicit enough in its intention.

So good practice is to always name boolean params that are options. The only exception to pass positional param as boolean is when this boolean value has a strong meaning that can be easily deduced by the function name, which is not the case here.